src/release-pr.ts

```diff
@@ -274,24 +304,60 @@ export class ReleasePR {
     });
   }
 
+  // Override this method to modify the pull request title
+  protected async buildPullRequestTitle(
+    version: string,
+    includePackageName: boolean
+  ): Promise<string> {
+    return includePackageName
+      ? `chore: release ${this.packageName} ${version}`
+      : `chore: release ${version}`;
+  }
+
+  // Override this method to detect the release version from code (if it cannot be
+  // inferred from the release PR head branch)
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected detectReleaseVersionFromTitle(title: string): string | undefined {
```

Comment:
@bcoe I recall your aversion to the instability of labels to store the version - I'd argue that the PR title is just as fragile - it could be changed after the last release-please run but before merging. It can also be changed after the PR has already been merged

@chingor13 is this here in support of keeping the version name out of the branch? If so, i think we should stick with just the manifest approach - I promise I'll have that up for review soon!

Reply:
This is fallback to support versionless branch names (until we implement the code detection).

Reply:
do we need versionless branch names right now? i.e. is there a need outside the aggregate PR to keep version names out of branches?

Reply:
On of our Java repos has a edge case need for this.

Reply:
If this unblocks @chingor13's work on Java, I'm tempted to run with this approach for the time being.

An additional concern I had with labels, is that our forking GitHub actions have permission to open PRs, but not to label.  So using labels had an additional permission headache.

I like that the lookup logic has been encapsulated and tested.